test/index.js

```diff
@@ -174,4 +176,81 @@ describe('stylelint-webpack-plugin', function () {
         });
     });
   });
+
+  context('lintDirtyModulesOnly flag is enabled', function () {
+    it('skips linting on initial run', function () {
+      var config = {
+        context: './test/fixtures/test3',
+        entry: './index',
+        plugins: [
+          new StyleLintPlugin({
+            configFile: configFilePath,
+            quiet: true,
+            lintDirtyModulesOnly: true
+          }),
+          new webpack.NoErrorsPlugin()
+        ]
+      };
+
+      return pack(assign({}, baseConfig, config))
+        .then(function (stats) {
+          expect(stats.compilation.errors).to.have.length(0);
+        });
+    });
+    it('lints only changed files in watch mode', function (done) {
+      this.timeout(5000);
+      var context = path.resolve(__dirname, 'fixtures/lint-dirty-files');
+      var config = {
+        context: context,
+        entry: './index',
+        plugins: [
+          new StyleLintPlugin({
+            configFile: configFilePath,
+            lintDirtyModulesOnly: true
+          })
+        ]
+      };
+
+      var destSass = context + '/test-tmp.scss';
+      var destJS = context + '/index.js';
+      fsExtra.copySync(context + '/initial-index.js', destJS);
+      fsExtra.copySync(context + '/initial-bad.scss', destSass);
+      var stop = watch(assign({}, baseConfig, config), watchCallback);
+      var runsCount = 0;
+
+      function watchCallback(err, stats) {
+        if (err) {
+          return done(err);
+        }
+
+        // First watch run doesn't have any information in stats yet.
+        // So, starting from 2nd.
+        if (runsCount === 1) {
+          // Check that there are no errors on initial run.
+          expect(stats.compilation.errors).to.have.length(0);
+          expect(stats.compilation.warnings).to.have.length(0);
+
+          // Trigger watch build by updating JS file, should have no errors in next step
+          fsExtra.copy(context + '/updated-index.js', destJS);
+        } else
+        if (runsCount === 2) {
+          // Check that on JS file change styleling is not triggered.
+          expect(stats.compilation.errors).to.have.length(0);
```

Comment:
@JaKXz, initially, I had separate test for testing for non-style file change. But seems like it's not possible to run multiple webpack watch instances simultaneously as tests are executed in parallel mode. So, I had to squash all watch cases into one.